HangMan/src/hangman.rs

```diff
@@ -0,0 +1,139 @@
+use std::vec;
+use dioxus::prelude::*;
+
+//component for managing the image of the hangman
+#[inline_props]
+pub fn Hangman(cx:Scope, stage: usize, reset_image: bool) -> Element {
+    
+    let css_status = use_ref(cx, || {vec![true;11]});    
+
+    //changes the hidden status of the next image in queque when the user makes a guess
+    use_memo(cx, stage, |stage| {
+        
+        css_status.with_mut(|current_list| {
+            if stage >= current_list.len() {
+                panic!("SHOULDNT ALLOW THE USER TO MAKE MORE THAN 10 GUESSES");
+            } else {
+                current_list[stage] = false;
+            }
+        })   
+    });
+
+    use_memo(cx, reset_image, |_| {
```

Comment:
I would try to avoid dependancies like reset_image that just serve as a signal for when to do an action. This callback will only run when reset_image changes not when reset_image is true which can make the behavior of the code confusing